blaze-server/src/main/scala/org/http4s/server/blaze/WSFrameAggregator.scala

```diff
@@ -0,0 +1,101 @@
+package org.http4s.server.blaze
+
+import org.http4s.blaze.pipeline.MidStage
+import org.http4s.blaze.util.Execution._
+import org.http4s.util
+import org.http4s.websocket.WebsocketBits._
+
+import scala.concurrent.{Future, Promise}
+import scala.collection.mutable.ArrayBuffer
+import scala.util.{Failure, Success}
+
+import java.net.ProtocolException
+
+private class WSFrameAggregator extends MidStage[WebSocketFrame, WebSocketFrame] {
+
+  def name: String = "WebSocket Frame Aggregator"
+
+  private var queue = new ArrayBuffer[WebSocketFrame]
+  private var size = 0
```

Comment:
Would this be any easier to maintain if it were a case class that held a buffer and the size, and adding to the buffer added to the size, and clearing it zeroed the size?  It seems like those mutations must happen together.

Reply:
Yes, that sounds like a good change.